### ISSUE TITLE
dotnet-runtime-deps-2.1 does not install all prerequisites (libcurl4 and libkrb5-3 are missing)

### ISSUE THREAD

**Issue by loop-evgeny** [dotnet/corefx#35409]
When trying to use `HttpWebRequest` in a self-contained .NET Core 2.1.8 application running on Ubuntu 18.04 I got the following exception:

```
-> Unable to load DLL 'System.Net.Http.Native': The specified module or one of its dependencies could not be found. (Exception from HRESULT: 0x8007007E) ---> The type initializer for 'HttpInitializer' threw an exception. ---> The type initializer for 'Http' threw an exception. ---> The type initializer for 'System.Net.Http.CurlHandler' threw an exception

at Interop.Http.GetSslVersionDescription()
at Interop.HttpInitializer..cctor() --->
at Interop.Http..cctor() --->
at Interop.Http.GetSupportedFeatures()
at System.Net.Http.CurlHandler..cctor() --->
at System.Net.HttpWebRequest.GetResponse()
```

This was fixed by installing the packages listed at https://docs.microsoft.com/en-us/dotnet/core/linux-prerequisites?tabs=netcore2x under "Ubuntu".

The machine had the dotnet-runtime-deps-2.1 APT package installed and I expected that to take care of installing all the native dependencies, but it depends only on the following:

```
$ apt-cache depends dotnet-runtime-deps-2.1
dotnet-runtime-deps-2.1
  Depends: libgcc1
  Depends: libssl1.0.0
  Depends: liblttng-ust0
  Depends: libc6
  Depends: libicu60
  Depends: zlib1g
  Depends: libstdc++6
  Depends: libgssapi-krb5-2
```
Note that libcurl4 and libkrb5-3 are missing. I would expect those to be included as well if they're required.

**Comment by karelz** [comment]
libcurl on .NET Core 2.1+ is optional. You will get the exception above, only if you opt-out from SocketsHttpHandler.
Having to install more dependencies in the case of opt-out sounds reasonable to me.

Closing as by design.

**Comment by loop-evgeny** [comment]
I think there are several issues remaining here:

1) I'm not sure what you mean by "opt-out from SocketsHttpHandler". I don't remember doing anything like that.
2) Even if this behaviour is intentional the error message is very bad. Is the user really supposed to know that "Unable to load DLL 'System.Net.Http.Native': The specified module or one of its dependencies could not be found" means "install libcurl4"?
3) Even if this behaviour is intentional it seems to be inconsistent with the documentation, which says "Ubuntu distributions require the following libraries installed: ... libcurl4" When the documentation says a package is *required* and there is a package named "dotnet-runtime-deps-2.1", don't you think it's a bit misleading that installs *most*, but not *all* of the "required" dependencies?
4) What about libkrb5-3?

**Comment by karelz** [comment]
1. Providing a minimal repro may help with that ... do you use other libraries? Are you able to reproduce it on other machines?
2. While true, given that we plan to remove libcurl support entirely post 3.0, I don't think it makes sense to improve the error messages at this point. Also taking into account the low number of reports on this.
3. I don't understand - which doc says libcurl4 is a pre-req?
4. That is Kerberos support AFAIK - it is optional only for people that need Kerberos (which is not everyone).

**Comment by loop-evgeny** [comment]
As it's been over a month since I raised this issue I don't remember anymore how I came across this problem. I can't repro it in a simple console EXE, but I definitely didn't set UseSocketsHttpHandler as described on https://docs.microsoft.com/en-us/dotnet/core/whats-new/dotnet-core-2-1 (I only learned about that now).

The doc that says libcurl4 is a pre-req is https://docs.microsoft.com/en-us/dotnet/core/linux-prerequisites?tabs=netcore2x (in the original issue). That says:

> Ubuntu distributions require the following libraries installed:
> 
> liblttng-ust0
> libcurl3 (for 14.x and 16.x)
> libcurl4 (for 18.x)
> libssl1.0.0
> libkrb5-3
> zlib1g
> libicu52 (for 14.x)
> libicu55 (for 16.x)
> libicu57 (for 17.x)
> libicu60 (for 18.x)
